oremda/pipeline.py

```diff
@@ -244,15 +270,18 @@ def deserialize_pipeline(obj, client):
     for _node in _nodes:
         node = OperatorNode(_node.get('id'))
 
-        _input_ports = _node.get('ports', {}).get('input', {})
-        _output_ports = _node.get('ports', {}).get('output', {})
+        _image_name = _node['image']
+        _ports = registry.ports(_image_name)
+        _input_ports = _ports.get('input', {})
+        _output_ports = _ports.get('output', {})
+        _queue_name = registry.name(_image_name)
         _params = _node.get('params', {})
-        _queue_name = _node['queue']
 
         input_ports = {}
         for name, port in _input_ports.items():
             port_type = validate_port_type(port.get('type'))
-            input_ports[name] = PortInfo(port_type, name)
+            required = bool_from_str(port.get('required', 'true'))
```

Comment:
Do you think I should get the `image.oremda_labels` to automatically try to convert label values from `str` to their corresponding type (`int`, `bool`, `float`, etc.)? That would simplify things like this, so we don't have to call `bool_from_str()`.

Reply:
Yes I think that would be the most appropriate place to do such conversions.
We should have some type of priority when trying to coerce the types (bool -> int -> float -> str ?)

Reply:
Sounds good, I'll do that in a separate PR after this gets merged.